src/pytest_split/plugin.py

```diff
@@ -181,6 +185,60 @@ def pytest_collection_modifyitems(
         return None
 
 
+def _reorganize_broken_up_ipynbs(group: "TestGroup", items: list) -> None:
```

Comment:
I feel that the notebook related functionality could deserve a dedicated module so that it would be clearly separated from the core logic, e.g. `ipynb_compatibility.py` could do.